src/app/data/devicesActions.js

```diff
@@ -0,0 +1,55 @@
+import {
+  getDocs, getDoc, doc,
+  addDoc, setDoc
+} from 'firebase/firestore';
+import { devicesCollection } from '../../app/firebase/firestoreConfig';
+import { setFormPrompt } from '../redux/form/formActions';
+
+const getAllDevices = async () => {
+  const response = await getDocs(devicesCollection);
+  const documents = response.docs;
+
+  const devices = [];
+  documents.forEach(item => {
+    devices.push({
+      id: item.id,
+      ...item.data()
+    })
+  })
+
+  return devices;
+}
+
+const getDeviceById = async id => {
+  const snap = await getDoc(doc(devicesCollection, id));
+  if (snap.exists()) {
+    return { ...snap.data(), id: id };
+  }
+}
```

Comment:
What if `snap.exists()` is false?

Reply:
My idea was if the snap doesn't exist, the function will return undefined (so I leave that case to be blank). But now as I see it creates confusion, I decided to change it to return null.